### PR TITLE
add kubectl version for helm and fix event bug

### PR DESCRIPTION
**Pull request by dellkeji**

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-bcs-saas?pullRequest=113) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-bcs-saas?pullRequest=113) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-bcs-saas?pullRequest=113) it.</sub>

File: bcs-app/backend/apps/application/instance_views.py [hunk 1]
```diff
@@ -961,11 +962,9 @@ def get(self, request, project_id, instance_id):
                 request, project_id, labels.get("io.tencent.paas.templateid"), curr_inst.namespace)
             category = curr_inst.category
         # 获取kind
-        flag, project_kind = self.get_project_kind(request, project_id)
-        if not flag:
-            return project_kind
+        project_kind = request.project.kind
         # 获取application
-        if category == DEPLOYMENT_CATEGORY:
+        if category == DEPLOYMENT_CATEGORY and project_kind == ProjectKind.MESOS.value:
```
Comment (jamesgetx) [inline bcs-app/backend/apps/application/instance_views.py]: 看上文代码，`DEPLOYMENT_CATEGORY`本身就是mesos吧，如果是k8s，应该是`K8SDEPLOYMENT_CATEGORY`

File: bcs-app/backend/bcs_k8s/bke_client/constants_bk.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# specific language governing permissions and limitations under the License.
+#
+import re
+from collections import OrderedDict
+
+# default kubectl version
+DEFAULT_KUBECTL_VERSION = '1.12.3'
+
+# KUBECTL VERSION
+KUBECTL_VERSION = OrderedDict({
```
Comment (jamesgetx) [inline bcs-app/backend/bcs_k8s/bke_client/constants_bk.py]: web_console模块也用到类似的KUBECTLD_VERSION，看是否有方式合并下这个配置

File: bcs-app/backend/utils/basic.py [hunk 3]
```diff
@@ -98,3 +98,15 @@ def getitems(obj, items, default=None):
         return reduce(lambda x, i: x[i], items, obj)
     except (IndexError, KeyError, TypeError):
         return default
+
+
+def get_bcs_component_version(cluster_version, bcs_component_version_info, default_version):
```
Comment (jamesgetx) [inline bcs-app/backend/utils/basic.py]: web_console定义了`get_kubectld_version`方法，看下是否可以抽取出来复用

File: bcs-app/backend/bcs_k8s/bke_client/constants.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
```
Comment (ifooth) [inline bcs-app/backend/bcs_k8s/bke_client/constants.py]: bke_client -> 这个要不一次改成bcs, bke这个后面应该不会用了

File: bcs-app/backend/utils/basic.py (hunk 3 above)
Comment (ifooth) [inline bcs-app/backend/utils/basic.py]: 这个web-console也封装了一次，要不要统一下

File: bcs-app/backend/apps/application/instance_views.py (hunk 1 above)
Comment (dellkeji) [reply bcs-app/backend/apps/application/instance_views.py]: 这个是为了兼容前端传递的类型

File: bcs-app/backend/bcs_k8s/bke_client/constants_bk.py (hunk 2 above)
Comment (dellkeji) [reply bcs-app/backend/bcs_k8s/bke_client/constants_bk.py]: 这个抽象一层处理

File: bcs-app/backend/utils/basic.py (hunk 3 above)
Comment (dellkeji) [reply bcs-app/backend/utils/basic.py]: 统一

File: bcs-app/backend/bcs_k8s/bke_client/constants.py (hunk 4 above)
Comment (dellkeji) [reply bcs-app/backend/bcs_k8s/bke_client/constants.py]: 这个这次先不变动，有好几个地方需要变动，需要多测试一下，保证无异常；先单独记录下

**Pull request merged**